src/internal/m365/collection/site/collection.go

```diff
@@ -366,3 +358,26 @@ func (sc *Collection) handleListItems(
 	sc.stream <- item
 	progress <- struct{}{}
 }
+
+func finishPopulation(
+	ctx context.Context,
+	stream chan data.Item,
+	su support.StatusUpdater,
+	fullPath path.Path,
+	metrics support.CollectionMetrics,
+) {
+	close(stream)
```

Comment:
Quick question unrelated to this PR:

If I do a backup with both `lists` and `pages` as selected categories, and say  `Items` is called for `lists` first. After finishing its going to close the stream. Given pages and lists both share the same object instance `sc`, when `Items` will be called for `pages` next, it's going to error out on closed channel in `retrievePages`. Is my understanding correct?

Reply:
yeah. that will be a problem.

Should we change `Items` to below:
```
stream := make(chan data.Item)
go sc.streamItems(ctx, errs, stream)
return stream
```

Reply:
Approach sounds good. You'd probably need a buffered channel like we have today, i.e. `make(chan data.Item, collectionChannelBufferSize)`.